config/check_dependencies.py

```diff
@@ -0,0 +1,169 @@
+#
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+#  SPDX-License-Identifier: BSD-3-Clause
+
+import getopt
+import json
+import jsonschema
+import sys
+import os
+
+from os.path import exists
+
+filename = "check_dependencies.py"
+
+COMPATIBLE_LICENSES = [
+    "MIT",
+    "GPLv2",
+    "The MIT License",
+    "Apache-2.0",
+    "Apache-1.1",
+    "PSF-2.0",
+    "BSD",
+    "BSD-2-Clause",
+    "BSD-3-Clause",
+    "GNU LGPL",
+    "GNU LGPL-2.1",
+    "GNU LGPL-3.0",
+    "CDDL-1.1",
+    "EDL-1.0",
+    "MPL-2.0-no-copyleft-exception",
+    "EPL-1.0",
+    "The JSON License"
+]
+
+
+class ComponentValidator:
+    def __init__(self, allow_component, sbom_component):
+        self.allow = allow_component
+        self.sbom = sbom_component
+
+        self.errors = []
+
+    def validate(self, product=False):
+        self.__name()
+        self.__licenses()
+        if not product:
+            self.__version()
+        return self.errors
+
+    def __name(self):
+        sbom_name = "{}:{}".format(self.sbom["group"], self.sbom["name"])
+        if self.allow["moduleName"] != sbom_name:
+            self.append_error("name", self.allow["moduleName"], sbom_name)
+
+    def __version(self):
+        if self.allow["moduleVersion"] != self.sbom["version"]:
+            self.append_error("version", self.allow["moduleVersion"], self.sbom["version"])
+
+    def __licenses(self):
+        if 'moduleLicense' in self.allow["actualLicense"]:
+            allow_license = self.allow["moduleLicense"]
+        else:
+            allow_license = self.allow["actualLicense"]
+
+        prepared_license_list = [lic.lower() for lic in COMPATIBLE_LICENSES]
+        if not (any([allow_license.lower() == lic for lic in prepared_license_list])):
+            self.append_error("license", COMPATIBLE_LICENSES, allow_license)
+
+    def append_error(self, key, expect, current):
+        self.errors.append("Component '{}':'{}' expects (one of) '{}' but was '{}'."
+                           .format(self.allow["moduleName"], key, expect, current))
+
+
+def read_json(file_path):
+    if not exists(file_path):
+        print("{} does not exist.".format(file_path))
+        sys.exit(2)
+
+    with open(file_path) as f:
+        return json.load(f)
+
+
+def compare_license_files(allowlist_path, sbom_path, allowschema_path):
+    allow_json = read_json(allowlist_path)
+    sbom_json = read_json(sbom_path)
+    schema_json = read_json(allowschema_path)
+    found_errors = []
+    found_warnings = []
+
+    # check for correct schema first
+    try:
+        jsonschema.validate(allow_json, schema_json)
+    except jsonschema.exceptions.ValidationError as error:
+        found_errors.append(f"allowlist does not comply to the schema...\n{error.message}")
+
+    # dependencies
+    allowed_packages = [item["moduleName"] for item in allow_json["allowedLicenses"]]
+    sbom_packages = ["{}:{}".format(item["group"], item["name"]) for item in sbom_json["components"]]
```

Comment:
Wouldn't it be easier and less complex, to transform them once into a dict each, with the module name as key. Here you could create warning, when a component is listed twice. 

For the diff you could use the .keys()

and for the actual comparison you would iterate through one dict and get the matching entry in the other one by simple accessing it

Reply:
The list generation is more complex but afterwards it only a string comparison

Reply:
But 
```
for component in sbom_json["components"]:
        # component["license"] = metadata(component["name"])["license"]
        for allowed_component in allow_json["allowedLicenses"]:

            name = allowed_component["moduleName"] == "{}:{}".format(component["group"], component["name"])
            if name:
                validator = ComponentValidator(allowed_component, component)
                ...
```

is way more complex than 

```
for name, sbom_entry in sbom_packages.items():
        if name in allow_packages:
                validator = ComponentValidator(allow_packages[name], sbom_entry)
                ...
```

Reply:
Keys() are unique. So we have to check multiple modules the other way